ts/session/medium_group/ratchet.ts

```diff
@@ -269,11 +276,23 @@ async function decryptWithSenderKeyInner(
     return null;
   }
 
-  // TODO: this might fail, handle this
-  const plaintext = await window.libloki.crypto.DecryptGCM(
-    messageKey,
-    ciphertext
-  );
-
-  return plaintext;
+  try {
+    const plaintext = await window.libloki.crypto.DecryptGCM(
+      messageKey,
+      ciphertext
+    );
+    return plaintext;
+  } catch (e) {
+    window.log.error('Got error during DecryptGCM():', e);
+    if (e instanceof DOMException) {
```

Comment:
Maybe you could re-throw this as `SenderKeyMissing` (if that seems to be the problem) and it would be handelled in a similar way in `contentMessage.ts`

Reply:
Amended